### PR TITLE
Refactor duck

### PR DESCRIPTION
**Pull request by IntelOrca**

### CODE REVIEW

**Comment by janisozaur** [comment]
does this include/replace #5300 ?

**Comment by IntelOrca** [comment]
no

**Comment by wolfreak99** [comment]
I just want to confirm that this is indeed a duck

**Comment by Joshwoo70** [comment]
*clicks duck* "quack"

File: src/openrct2/world/Duck.cpp [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+}
+
+void rct_duck::UpdateSwim()
+{
+    if (((gCurrentTicks + sprite_index) & 3) != 0) return;
+
+    uint32 randomNumber = scenario_rand();
+    if ((randomNumber & 0xFFFF) < 0x666)
+    {
+        if (randomNumber & 0x80000000)
```
Comment (duncanspumpkin) [inline src/openrct2/world/Duck.cpp]: In some of the peep code I would just convert this into a new `scenario_rand` call. I'm not fussed if you think its better to do it this way though.

File: src/openrct2/world/Duck.cpp [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+                }
+            }
+        }
+    }
+}
+
+void rct_duck::UpdateDrink()
+{
+    frame++;
+    if (DuckAnimationDrink[frame] == 0xFF)
```
Comment (duncanspumpkin) [inline src/openrct2/world/Duck.cpp]: Is there a reason these (drink and doubledrink) are not using the Util::CountOf function?

Comment (IntelOrca) [inline src/openrct2/world/Duck.cpp]: This is how the original assembly did it.
The difference is that 0xFF is used to indicate a change of animation where as the ones without are simple loops.
Of course it doesn't make any sense to do this when its got separate logic for each animation type.

**Pull request merged**